python/mim_control/dynamic_graph/wbc_graph.py

```diff
@@ -180,19 +186,18 @@ def plug(self, robot, base_position, base_velocity):
             dg.plug(velocity, imp.robot_velocity_sin)
 
         # Setting the actual quantities for the centroidal-pd controller.
-        # NOTE: Using the base as approximation for the com.
         dg.plug(
-            selec_vector(base_position, 0, 3),
-            self.wcom_pd_ctrl.actual_com_position_sin,
+            multiply_mat_vec(
+                self.dg_robot.signal('Jcom'),
+                velocity
+            ),
+            self.wcom_pd_ctrl.actual_com_velocity_sin
```

Comment:
Rewriting this section to use the actual com of the robot and not the base position. Also switch from using the base velocity to the com velocity. Note that there is no direct way to get vcom right now in sot-pinocchio, therefore doing the workaround with Jcom * dq. Note as well that Jcom * dq is in world frame, which is what we want.